path_data_test.go

```diff
@@ -75,6 +77,8 @@ func expectedMetadataKeys() map[string]struct{} {
 func TestVersionedKV_Data_Put(t *testing.T) {
 	b, storage := getBackend(t)
 
+	paths := []*framework.Path{pathData(b.(*versionedKVBackend))}
```

Comment:
I was debating between the two approaches for this and other plugins' tests. The problem with doing schema verification in separate tests is that it will introduce a lot of test code duplication.  Many of these tests are heavy with the setup logic. For example in order to verify the `GET` schema, we would need to duplicate the majority of `TestVersionedKV_Data_Get` which has ~60 lines of setting up the test (writing to `data/foo` and `metadata/foo`).

The other advantage of doing it inline is that we are verifying that the responses match the declared schema in all of the test scenarios, giving us better code coverage. For example, `"deletion_time"` is sometimes returned as an empty string, which is incongruous with`framework.TypeTime` and has to be represented as a `framework.TypeString` instead. If we only set up one test for the `GET` schema, it might not have caught this nuance.